### DISCUSSION TITLE
how complicated would it be to have a degraded snappymail mode for old browsers?

### DISCUSSION THREAD

**Discussion by yeupou** [the-djmaze/snappymail#1064] [General]
In some workplace, I am stuck with an old firefox version (around 52) and no update planned by the distant sysadmin for a while.

I understand that Snappymail started as an improved fork  using "downsized/simplified versions of scripts and has no support for Internet Explorer nor Edge Legacy. Supported are: [...]Firefox 78+".
But at this point, Snappymail is no longer just a snappy alternative but a main app. 

I'd rather have Snappymail in degraded mode instead of  installing another webmail for very rare (but existing) situation.

The currently version with Firefox 52.x is stuck at login. The login page stays with no content and non ending circular loading image. 
Would it require much work to be able to login and do basic activities even with old version, scriptless (maybe with a big red bar on top warning it is running in degraded mode)?

**Comment by randolf** [comment]
You can run an older version from a different virtual host on your web server, and then instruct your users with older web browsers to use that hostname instead.

To automate redirection to the alternative hostname, you might be able to set up Apache Rewrite rules to do this, or, failing that, create a mod_perl2 script at an early handler stage that uses the HTTP::BrowserDetect module to detect the web browser and version, and conditionally redirect as need.

**Comment by the-djmaze** [comment]
A lot of the code uses ecmascript 2018+
This means a transpiler needs to convert all code to a very old ecmascript version.
And there need to be many polyfills to cover crashes.

This is hardly possible, so basically you are out of luck (not to.mention all old browser security holes themselves)
